src/main/kotlin/dev/slne/surfModerationTools/listener/PlayerActionListener.kt

```diff
@@ -0,0 +1,98 @@
+package dev.slne.surfModerationTools.listener
+
+import com.github.benmanes.caffeine.cache.Caffeine
+import dev.slne.surf.surfapi.core.api.messages.adventure.sendText
+import dev.slne.surfModerationTools.utils.FreezeManager
+import org.bukkit.entity.Player
+import org.bukkit.event.EventHandler
+import org.bukkit.event.Listener
+import org.bukkit.event.block.BlockBreakEvent
+import org.bukkit.event.block.BlockPlaceEvent
+import org.bukkit.event.entity.EntityDamageByEntityEvent
+import org.bukkit.event.player.PlayerInteractEntityEvent
+import org.bukkit.event.player.PlayerInteractEvent
+import org.bukkit.event.player.PlayerMoveEvent
+import org.bukkit.event.player.PlayerQuitEvent
+import java.util.UUID
+import java.util.concurrent.TimeUnit
+
+class PlayerActionListener : Listener {
+
+    private val messageCooldown = Caffeine.newBuilder()
+        .expireAfterWrite(5, TimeUnit.SECONDS)
+        .build<UUID, Long>()
+
+    private fun sendFrozenMessage(player: Player, message: String) {
+        if ((messageCooldown.getIfPresent(player.uniqueId) ?: 0) < System.currentTimeMillis()) {
+            player.sendText {
+                appendPrefix()
+                error(message)
+            }
+            messageCooldown.put(player.uniqueId, System.currentTimeMillis() + 5_000)
+        }
+    }
+
+    @EventHandler
+    fun onPlayerMove(event: PlayerMoveEvent) {
+        val player = event.player
+        if (FreezeManager.isPlayerFrozen(player)) {
```

Comment:
```suggestion
    fun onPlayerMove(event: PlayerMoveEvent) {
        val player = event.player
        
        if(!event.hasChangedBlock()) {
            return
        }
        
        if (FreezeManager.isPlayerFrozen(player)) {
```

kleine performance verbesserung